index.js

```diff
@@ -64,15 +64,15 @@ function TChannel(options) {
 
 	this.endpoints = Object.create(null);
 	this.destroyed = false;
-
-	// TODO do not create a tcp server in the constructor
-	// Doing IO in a constructor is bad. Move into a method.
+	// to provide backward compatibility. 
+	this.listening = this.options.listening === false ?
+	  false : true;
 	this.serverSocket = new net.createServer();
-	// TODO do not call listen by default. We might want to
-	// seperate allocating a server from calling listen on it.
-	// We can module.exports a `createTchannel()` helper that
-	// calls listen() by default.
-	this.serverSocket.listen(this.port, this.host);
+
+	if (this.listening) {
+		this.createTchannel();
+	}
+
 	this.serverSocket.on('listening', function () {
```

Comment:
if `this.listening` is false then `this.serverSocket` is undefined and this will throw. We need to refactor this as well.